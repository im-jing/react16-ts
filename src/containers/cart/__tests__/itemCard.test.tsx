import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import ItemCard, { Props } from "../itemCart";

const testProps: Props = {
  id: 3,
  checkStatus: true,
  unitPrice: 1000,
  itemContent: "desk",
};

describe("itemCard component", () => {
  it("should render the itemCard component correctly", () => {
    const { container, debug, getByRole } = render(<ItemCard {...testProps} />);
    // debug(getByRole('item-card'))

    expect(getByRole("item-card")).toHaveClass("item-card");
    expect(container).toMatchSnapshot();
  });
});
