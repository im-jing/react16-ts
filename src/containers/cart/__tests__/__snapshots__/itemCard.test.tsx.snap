// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemCard component should render the itemCard component correctly 1`] = `
<div>
  <div
    class="item-card"
    role="item-card"
  >
    <div
      class="checkbox-wrap"
    >
      <label
        class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
      >
        <span
          class="ant-checkbox ant-checkbox-checked"
        >
          <input
            checked=""
            class="ant-checkbox-input"
            type="checkbox"
            value="3"
          />
          <span
            class="ant-checkbox-inner"
          />
        </span>
      </label>
      <span>
        desk
      </span>
    </div>
    <p
      class="item-info"
    >
      ï¿¥
      1000
    </p>
  </div>
</div>
`;
